viewer/packages/360-images/src/Image360Entity.ts

```diff
@@ -76,15 +79,50 @@ export class Image360Entity {
     if (this._imageContainer === undefined) {
       return;
     }
-    this._imageContainer.visible = false;
+    const imageContainer = await this._imageContainer;
+    imageContainer.visible = false;
+  }
+
+  public async load360Image(): Promise<void> {
+    if (this._imageContainer !== undefined) {
+      return;
+    }
+
+    this._imageContainer = this._imageProvider
+      .get360ImageFiles(this._image360Metadata)
+      .then(faces => this.createImage360VisualizationObject(faces));
+    const imageContainer = await this._imageContainer;
+    imageContainer.applyMatrix4(this._transform);
+    this._sceneHandler.addCustomObject(imageContainer);
+    imageContainer.visible = false;
+  }
+
+  public async unload360Image(): Promise<void> {
+    if (this._imageContainer === undefined) {
+      return;
+    }
+
+    const imageContainer = await this._imageContainer;
+
+    this._sceneHandler.removeCustomObject(imageContainer);
+    const imageContainerMaterial = imageContainer.material;
+    const materials =
+      imageContainerMaterial instanceof THREE.Material ? [imageContainerMaterial] : imageContainerMaterial;
+
+    materials
+      .map(material => material as THREE.MeshBasicMaterial)
+      .forEach(material => {
+        material.map?.dispose();
+        material.dispose();
+      });
+
```

Comment:
Is this really the easiest way of disposing material? Would hope that the mesh did this automatically :/

Reply:
It sadly cannot safely do this as two separate meshes can share the same material and there is no reference counting or any smart stuff going on afaik. Same goes for textures, as the texture can be in use in other materials etc.